cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ClusterPartitionState.java

```diff
@@ -112,25 +115,26 @@ protected void populateKafkaPartitionState(Set<PartitionInfo> underReplicatedPar
         }
         for (PartitionInfo partitionInfo : _kafkaCluster.partitionsForTopic(topic)) {
           int numInsyncReplicas = partitionInfo.inSyncReplicas().length;
+          boolean hasOfflineReplicas = partitionInfo.offlineReplicas().length != 0;
           boolean isURP = numInsyncReplicas != partitionInfo.replicas().length;
+          boolean isOffline = partitionInfo.leader() == null;
+
           if (numInsyncReplicas < minInsyncReplicas) {
             underMinIsrPartitions.add(partitionInfo);
           }
-          if (isURP || verbose) {
-            boolean hasOfflineReplica = partitionInfo.offlineReplicas().length != 0;
-            if (hasOfflineReplica) {
-              partitionsWithOfflineReplicas.add(partitionInfo);
-            }
-            boolean isOffline = partitionInfo.inSyncReplicas().length == 0;
-            if (isOffline) {
-              offlinePartitions.add(partitionInfo);
-            } else if (isURP) {
-              underReplicatedPartitions.add(partitionInfo);
-            } else {
-              // verbose -- other
-              otherPartitions.add(partitionInfo);
-            }
+          if (hasOfflineReplicas) {
+            partitionsWithOfflineReplicas.add(partitionInfo);
+          }
+          if (isURP) {
```

Comment:
If it is offline, I think we don't want to add it into URP. This is to avoid verbose output.

Reply:
Updated ✅